src/Microsoft.AspNet.Mvc.Core/Formatters/MediaTypeEncoding.cs

```diff
@@ -0,0 +1,111 @@
+        public static Encoding GetEncoding(StringSegment mediaType)
+        {
+            var charset = GetCharset(mediaType);
+            return GetEncodingFromCharset(charset);
+        }
+
+        /// <summary>
+        /// Gets the <see cref="Encoding"/> for the given <see cref="mediaType"/> if it exists.
+        /// </summary>
+        /// <param name="mediaType">The mediaType from which to get the charset parameter.</param>
+        /// <returns>The <see cref="Encoding"/> of the mediaType if it exists or a <see cref="StringSegment"/> without value if not.</returns>
+        public static Encoding GetEncoding(string mediaType)
+        {
+            var charset = GetCharset(new StringSegment(mediaType));
+            return GetEncodingFromCharset(charset);
+        }
+
+        /// <summary>
+        /// Gets the charset parameter of the given <paramref name="mediaType"/> if it exists.
+        /// </summary>
+        /// <param name="mediaType">The mediaType from which to get the charset parameter.</param>
+        /// <returns>The charset of the mediaType if it exists or a <see cref="StringSegment"/> without value if not.</returns>
+        public static StringSegment GetCharset(StringSegment mediaType)
+        {
+            try
+            {
+                var mt = MediaTypeHeaderValue.Parse(mediaType.Value);
+                return new StringSegment(mt.Charset);
+
+            }
+            catch (Exception)
+            {
+
+                return new StringSegment();
+            }
+        }
+
+        /// <summary>
+        /// Replaces the encoding of the given <paramref name="mediaType"/> with the provided
+        /// <paramref name="encoding"/>.
+        /// </summary>
+        /// <param name="mediaType">The media type whose encoding will be replaced.</param>
+        /// <param name="encoding">The encoding that will replace the encoding in the <paramref name="mediaType"/></param>
+        /// <returns>A media type with the replaced encoding.</returns>
+        public static string ReplaceEncoding(string mediaType, Encoding encoding)
+        {
+            return ReplaceEncoding(new StringSegment(mediaType), encoding);
+        }
+
+        /// <summary>
+        /// Replaces the encoding of the given <paramref name="mediaType"/> with the provided
+        /// <paramref name="encoding"/>.
+        /// </summary>
+        /// <param name="mediaType">The media type whose encoding will be replaced.</param>
+        /// <param name="encoding">The encoding that will replace the encoding in the <paramref name="mediaType"/></param>
+        /// <returns>A media type with the replaced encoding.</returns>
+        public static string ReplaceEncoding(StringSegment mediaType, Encoding encoding)
+        {
+            var charset = GetCharset(mediaType);
+
+            if (GetEncodingFromCharset(charset) == encoding)
+            {
+                return mediaType.Value;
+            }
```

Comment:
It's just an optimization in case you pass the same encoding as the one on the media type